lib/reek/ast/sexp_extensions/module.rb

```diff
@@ -45,28 +45,45 @@ module ModuleNode
       # Utility methods for :class nodes.
       module ClassNode
         include ModuleNodeBase
+
         def superclass() children[1] end
+        def ancestor() superclass end
       end
 
       # Utility methods for :casgn nodes.
       module CasgnNode
         include ModuleNodeBase
 
         def defines_module?
-          return false unless value
-          call = case value.type
-                 when :block
-                   value.call
-                 when :send
-                   value
-                 end
+          call = module_definition
           call && call.module_creation_call?
         end
 
+        def module_definition
+          return nil unless value
```

Comment:
Yep, that's the problem with our codebase. Things like that are all over the place and you have no idea why 😆 
I normally then open pry and debug my way through which is time consuming and tedious.
Unfortunately AST and especially Ruby's AST is not self-explanatory or trivial.
My goal is to explain all those place in our code because this trips you up all the time.
So let's start here ;)